From 8291259cd6ed6865728e1f8b73830f8fe420e540 Mon Sep 17 00:00:00 2001
From: Mooffie <mooffie@gmail.com>
Date: Tue, 13 Dec 2016 17:06:12 +0200
Subject: [PATCH 09/13] extfs: hp48: documentation: explain the reply to the
 DIRECTORY command.

It was not at all easy to find a sample reply on the internet (the one found
was in a screenshot actually), so we record it here.
---
 src/vfs/extfs/helpers/hp48+.in | 24 +++++++++++++++++++++++-
 1 file changed, 23 insertions(+), 1 deletion(-)

diff --git a/src/vfs/extfs/helpers/hp48+.in b/src/vfs/extfs/helpers/hp48+.in
index b67fce5..ef136ec 100644
--- a/src/vfs/extfs/helpers/hp48+.in
+++ b/src/vfs/extfs/helpers/hp48+.in
@@ -17,7 +17,7 @@
 #                         ascii (editable on the pc)
 #      - start the server on the HP48: <left shift> - i/o - srvr - serve
 #        or the shortcut <right shift> - <right arrow>
-#      - on MC's commandline enter "cd hp48:"
+#      - on MC's commandline enter "cd hp48://"
 #
 # Make sure you have kermit installed and that it's using the right serial
 # device by changing /dev/ttyXX on the next line
@@ -48,6 +48,28 @@ hp48_retsize()
 printf "%d" "$2" 2>/dev/null
 }
 
+#
+# Parses the reply to the DIRECTORY command.
+#
+# Here's an example reply (taken from [1][2]):
+#
+#    { HOME } 105617
+#    STRAY 185.5 Directory 29225
+#    YEN 30.5 Program 53391
+#    JYTLIGHT 21848.5 String 62692
+#    IOPAR 37.5 List 61074
+#
+# The meaning of the fields (according to [3][4]):
+#
+#    { Current_directory } Free_space
+#    Object_name Object_size_bytes Object_type Object_CRC
+#    ...
+#
+# [1] http://newarea48.tripod.com/kermit.html
+# [2] http://www.hpmuseum.org/forum/thread-4684.html
+# [3] https://groups.google.com/d/msg/comp.sys.hp48/bYTCu9K3k20/YWQfF--W3EEJ
+# [4] http://www.columbia.edu/kermit/hp48.html (also has a link to the HP's user manual).
+#
 hp48_parser()
 {
 HP48_DIRS=
-- 
2.9.3

